helix-core/src/main/java/org/apache/helix/monitoring/metrics/implementation/BaselineDivergenceGauge.java

```diff
@@ -0,0 +1,38 @@
+package org.apache.helix.monitoring.metrics.implementation;
+
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import org.apache.helix.monitoring.metrics.model.RatioMetric;
+
+
+/**
+ * Gauge of the difference (state and partition allocation) between the baseline and the best
+ * possible assignment.
+ */
+public class BaselineDivergenceGauge extends RatioMetric {
+  private static final double VALUE_NOT_SET = -1.0d;
```

Comment:
If not set, just return 0.0?

Reply:
I initially set it to 0.0. But making it consistent with the latencyGuage's init value -1L. I actually think this is just a one time value. I don't think it makes any difference. VALUE_NOT_SET means it could be out of range [0, 1].
Init value could be set to 0.0, but VALUE_NOT_SET may set -1.0 to be different.

Reply:
This is a gauge, so if we don't have a rebalance done for a while after the system starts, we will see -1 in the metric. Once it is persisted and visualized, how to explain divergence is -1?

And, which latency guage are you talking about? If it was -1, let change to 0, I think.

Reply:
Or, you need to find a way to stop data emitting when the value is VALUE_NOT_SET.